src/test/kotlin/no/nav/tilleggsstonader/sak/vilkår/stønadsvilkår/VilkårServiceIntegrasjonsTest.kt

```diff
@@ -86,6 +87,50 @@ internal class VilkårServiceIntegrasjonsTest : IntegrationTest() {
         assertVilkårErGjenbrukt(vilkårForBehandling, vilkårForRevurdering)
     }
 
+    @Test
+    internal fun `kopierVilkårsettTilNyBehandling - skal ikke kopiere vilkår som mangler periode`() {
+        val fagsak = testoppsettService.lagreFagsak(fagsak())
+        val behandling = testoppsettService.lagre(behandling(fagsak, status = BehandlingStatus.FERDIGSTILT))
+        val revurdering = testoppsettService.lagre(behandling(fagsak))
+        val barn1Ident = FnrGenerator.generer(LocalDate.now().minusYears(3))
+        val barnFørsteBehandling = barnRepository.insertAll(listOf(barn1Ident).tilBehandlingBarn(behandling))
+        val barnIdMap = barnService.gjenbrukBarn(behandling.id, revurdering.id)
+
+        opprettVilkårsvurderinger(behandling, barnFørsteBehandling, fom = null, tom = null).first()
+
+        vilkårService.kopierVilkårsettTilNyBehandling(
+            forrigeBehandlingId = behandling.id,
+            nyBehandling = revurdering,
+            barnIdMap = barnIdMap,
+        )
+
+        val vilkårForRevurdering = vilkårRepository.findByBehandlingId(revurdering.id)
+
+        assertThat(vilkårForRevurdering).isEmpty()
+    }
+
+    @Test
+    internal fun `kopierVilkårsettTilNyBehandling - skal ikke kopiere vilkår som er slettet`() {
+        val fagsak = testoppsettService.lagreFagsak(fagsak())
+        val behandling = testoppsettService.lagre(behandling(fagsak, status = BehandlingStatus.FERDIGSTILT))
+        val revurdering = testoppsettService.lagre(behandling(fagsak))
+        val barn1Ident = FnrGenerator.generer(LocalDate.now().minusYears(3))
+        val barnFørsteBehandling = barnRepository.insertAll(listOf(barn1Ident).tilBehandlingBarn(behandling))
+        val barnIdMap = barnService.gjenbrukBarn(behandling.id, revurdering.id)
+
+        opprettVilkårsvurderinger(behandling, barnFørsteBehandling, status = VilkårStatus.SLETTET).first()
+
+        vilkårService.kopierVilkårsettTilNyBehandling(
+            forrigeBehandlingId = behandling.id,
+            nyBehandling = revurdering,
+            barnIdMap = barnIdMap,
+        )
+
+        val vilkårForRevurdering = vilkårRepository.findByBehandlingId(revurdering.id)
+
+        assertThat(vilkårForRevurdering).isEmpty()
```

Comment:
Bra du skriver tester! Ser det er veldig mye oppsett som er likt mellom testene i denne klassen. Kunne noe av det vært trukket ut og gjenbrukt? Da blir det lettere å se hvilket oppsett som er relevant for hver enkel testcase, og det blir lettere å lage nye tester også.